### PR TITLE
Make Clippy Fierce

### PR DESCRIPTION
**Pull request by cwfitzgerald**
This PR does one primary thing and many secondary things. It's goal is to enable the shotgun approach to clippy warmings: enable them all, then disable the ooky ones. This approach has unearthed many warnings that I had no idea existed using features I had no idea existed.

Cargo warnings were also enabled. This forced the metadata to be fully flushed out and there be only a single version of any dependency. This sparked https://github.com/mystor/synstructure/pull/34 and the overriding of a couple of dependencies with git versions. I have specifically used the patch feature because we do not _depend_ on the git version, we just want all builds in the nova universe to use the git version.

Some code was made a bit more idiomatic.

### CODE REVIEW

File: src/shaderpack/mod.rs [hunk 1]
```diff
@@ -198,7 +202,7 @@ where
     // Iterate through the materials directory to find the useful files in the files with the needed extant
     for path in materials_folder {
         let full_path = path!("materials" | &path).into();
-        let ext = path.extension().and_then(|s| s.to_str());
+        let ext = path.extension().and_then(std::ffi::OsStr::to_str);
```
Comment (DethRaid) [inline src/shaderpack/mod.rs]: Does this name have to be fully qualified?

Comment (cwfitzgerald) [reply src/shaderpack/mod.rs]: No

**Comment by DethRaid** [comment]
Apparently Travis can't get all the dependencies it needs

@cwfitzgerald could you fix Travis, or disable Travis?

**Comment by cwfitzgerald** [comment]
The problem is that one of our tools doesn't build on this version of nightly. I'll make a adjustment PR and then rebase.

**Pull request merged**